pkg/apis/core/types.go

```diff
@@ -3330,6 +3330,16 @@ type LoadBalancerIngress struct {
 	Hostname string
 }
 
+// ServiceIPFamily represents the IP Family which the service will expose
+type ServiceIPFamily string
+
+const (
+	// IPv4Service indicates that this service will expose IPv4 IP and will select only endpoints that are IPv4
+	IPv4Service ServiceIPFamily = "IPv4"
```

Comment:
As you remoarked on slack - can rename this type and instances  to drop "Service".  Maybe put it next to Protocol ?

Reply:
do we need to do that for PR, or have it as a separate PR?

Reply:
either is fine :)